contributors/design-proposals/add-new-patchStrategy-to-clear-fields-not-present-in-patch.md

```diff
@@ -0,0 +1,340 @@
+Add new patchStrategy to clear fields not present in the patch
+=============
+
+Add tags `patchStrategy:"replaceKeys"`. For a given type that has the tag, all keys/fields missing
+from the request will be cleared when patching the object.
+For a field presents in the request, it will be merged with the live config.
+
+The proposal of Full Union is in [kubernetes/community#388](https://github.com/kubernetes/community/pull/388).
+
+| Capability | Supported By This Proposal | Supported By Full Union |
+|---|---|---|---|
+| Auto clear missing fields on patch | X | X |
+| Merge union fields on patch | X | X |
+| Validate only 1 field set on type |  | X |
+| Validate discriminator field matches one-of field |  | X |
+| Support non-union patchKey | X | TBD |
+| Support arbitrary combinations of set fields | X |  |
+
+## Use cases
+
+- As a user patching a map, I want keys mutually exclusive with those that I am providing to automatically be cleared.
+
+- As a user running kubectl apply, when I update a field in my configuration file,
+I want mutually exclusive fields never specified in my configuration to be cleared.
+
+## Examples:
+
+- General Example: Keys in a Union are mutually exclusive. Clear unspecified union values in a Union that contains a discriminator.
+
+- Specific Example: When patching a Deployment .spec.strategy, clear .spec.strategy.rollingUpdate
+if it is not provided in the patch so that changing .spec.strategy.type will not fail.
+
+- General Example: Keys in a Union are mutually exclusive. Clear unspecified union values in a Union
+that does not contain a discriminator.
+
+- Specific Example: When patching a Pod .spec.volume, clear all volume fields except the one specified in the patch.
+
+## Proposed Changes
+
+### APIs
+
+**Scope**:
+
+| Union Type | Supported |
+|---|---|---|
+| non-inlined non-discriminated union | Yes |
+| non-inlined discriminated union | Yes |
+| inlined union with [patchMergeKey](https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#strategic-merge-patch) only | Yes |
+| other inlined union | No |
+
+For the inlined union with patchMergeKey, we move the tag to the parent struct's instead of
+adding some logic to lookup the metadata in go struct of the inline union.
+Because the limitation of the latter is that the metadata associated with
+the inlined APIs will not be reflected in the OpenAPI schema.
+
+#### Tags
+
+old tags:
+
+1) `patchMergeKey`:
+It is the key to distinguish the entries in the list of non-primitive types. It must always be
+present to perform the merge on the list of non-primitive types, and will be preserved.
+
+2) `patchStrategy`:
+It indicates how to generate and merge a patch for lists. It could be `mrege` or `replace`. It is optional for lists.
+
+new tags:
+
+`patchStrategy: replaceKeys`:
+
+We introduce a new value `replaceKeys` for `patchStrategy`. It indicates the all the fields that
```

Comment:
It indicates **that** all fields needing to be preserved must **be** present in the patch.